docs/rackhd/event_notification.rst

```diff
@@ -1,105 +1,272 @@
-| obms.unassigned | one-shot  | Event occurs when node's `obms`  | Triggered when obm is          |
-|                 |           | field is unassigned.             | not associated with the node   |
-+-----------------+-----------+----------------------------------+--------------------------------+
-| obms.updated    | one-shot  | Event occurs when node's `obms`  | Triggered when node don't have |
-|                 |           | field is updated.                | obm settings                   |
-+-----------------+-----------+----------------------------------+--------------------------------+
-| accessible      | one-shot  | Event occurs when node telemetry | Triggered when any poller of a |
-|                 |           | OBM service (IPMI or SNMP) is    | node become accessible while   |
-|                 |           | accessible                       | the node's state is            |
-|                 |           |                                  | inaccessilbe or null           |
-+-----------------+-----------+----------------------------------+--------------------------------+
-| inaccessible    | one-shot  | Event occurs when node telemetry | Triggered when all pollers     |
-|                 |           | OBM service (IPMI or SNMP) is    | of a node become inaccessible. |
-|                 |           | inaccessible                     |                                |
-+-----------------+-----------+----------------------------------+--------------------------------+
+    node.discovered.information.57b15c51450be454180fa460.57b15c51450be454180fa460
 
+Graph event routing key:
+
+.. code-block:: REST
+
+    graph.started.information.35b15c51450be454180fabd.57b15c51450be454180fa460
+
+
+AMQP Routing Key Filter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All the events could be filtered by routing keys, for example:
+
+All services' heartbeat events:
+
+.. code-block:: Bash
+
+    $ sudo node sniff.js "on.events" "heartbeat.#"
+
+All nodes' discovered events:
+
+.. code-block:: Bash
+
+    $ sudo node sniff.js "on.events" "#.discovered.#"
+
+'sniff.js' is a tool located at https://github.com/RackHD/on-tools/blob/master/dev_tools/README.md
+
+
+Events via Hook
+~~~~~~~~~~~~~~~~~~~~
+
+URL could be registered by API below, and the filter conditions also could be specified in the payload
+
+.. code-block:: REST
+
+    curl -H "Content-Type: application/json" -X POST -d @payload.json <server>/api/2.0/hooks
+
+The payload.json format are as below:
+
+========= ====== ============ ============================================
+Attribute Type   Flags        Description
+========= ====== ============ ============================================
+url       String **required** The hook url that events are notified to.
+name      String **optional** Any name user specified.
+filters   Array  **optional** The conditions of which events are notified.
+========= ====== ============ ============================================
+
+Example of payload
```

Comment:
Based on the example, filters are an array of JSON objects with some predefined fields. In the example we see `type`, `typeId`, `action`, `severity`, and `nodeId`. How should someone know what are the possible filter values to search for under type, typeId, action, or severity? Are there any other key's that could be used for the filter